### PR TITLE
Implements Trading Economics Data 

### LINKED ISSUE #3200: Implements TradingEconomics Custom Data
#### Expected Behavior
Lean includes custom data from [TradingEconomics](https://tradingeconomics.com/).

#### Actual Behavior
[TradingEconomics](https://tradingeconomics.com/) data is not supported.

#### Potential Solution
Implements custom data for [TradingEconomics](https://tradingeconomics.com/). Here is the link to API: http://docs.tradingeconomics.com/#events

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

### PR DESCRIPTION
**Pull request by AlexCatarino**
#### Description
Implements Trading Economics Data Downloader
- Implements an interface for all Trading Economics Data
 - Adds Trading Economics Calendar Downloader and program to be used in the Toolbox.
 - Adds Trading Economics Indicators Downloader and program to be used in the Toolbox.
 - Adds Trading Economics Earnings Downloader and program to be used in the Toolbox.

- Adds TradingEconomicsCalendarIndicatorAlgorithm  …
 - Adds C# and Python version of `TradingEconomicsCalendarIndicatorAlgorithm` to show how `TradingEconomics` data can be added to an algorithm.

#### Related Issue
Close #3200 

#### Motivation and Context
Add new alternative data sources to Lean.

#### Requires Documentation Change
Be added to the Data docs.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Data/Custom/TradingEconomics/TradingEconomicsSymbolJsonConverter.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        /// Reads the JSON representation of the object.
+        /// </summary>
+        /// <param name="reader">The <see cref="T:Newtonsoft.Json.JsonReader"/> to read from.</param><param name="objectType">Type of the object.</param><param name="existingValue">The existing value of object being read.</param><param name="serializer">The calling serializer.</param>
+        /// <returns>
+        /// The object value.
+        /// </returns>
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var value = (reader.Value as string) ?? string.Empty;
+            return new Symbol(SecurityIdentifier.Empty, value);
```
Comment (mchandschuh) [inline Common/Data/Custom/TradingEconomics/TradingEconomicsSymbolJsonConverter.cs]: What kind of symbols are these? Will we ever need to correlate it back to a security tradable in LEAN?

File: ToolBox/TradingEconomicsDataDownloader/TradingEconomicsCalendarDownloaderProgram.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+{
+    public static class TradingEconomicsCalendarDownloaderProgram
+    {
+        /// <summary>
+        /// Trading Economics Calendar Downloader Toolbox Project For LEAN Algorithmic Trading Engine.
+        /// </summary>
+        public static void TradingEconomicsCalendarDownloader()
+        {
+            // Load settings from config.json
+            var dataDirectory = Config.Get("data-directory", "../../../Data");
```
Comment (mchandschuh) [inline ToolBox/TradingEconomicsDataDownloader/TradingEconomicsCalendarDownloaderProgram.cs]: no need to change, but I believe this is the same as `Global.DataFolder`

File: ToolBox/TradingEconomicsDataDownloader/TradingEconomicsDataDownloader.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                    var response = await client.GetAsync(Uri.EscapeUriString(url));
+
+                    response.EnsureSuccessStatusCode();
+
+                    return await response.Content.ReadAsStringAsync();
+                }
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Error at HttpRequester with msg: {e}", e);
```
Comment (mchandschuh) [inline ToolBox/TradingEconomicsDataDownloader/TradingEconomicsDataDownloader.cs]: this `e` in the error message will end up being the entire thing, including the stack trace. Might want to use the `e.Message` property here and then wrap `e` as the inner like you have done.

File: ToolBox/TradingEconomicsDataDownloader/TradingEconomicsEarningsDownloaderProgram.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public static class TradingEconomicsEarningsDownloaderProgram
+    {
+        /// <summary>
+        /// Trading Economics Earnings Downloader Toolbox Project For LEAN Algorithmic Trading Engine.
+        /// </summary>
+        public static void TradingEconomicsEarningsDownloader()
+        {
+            // Load settings from config.json
+            var dataDirectory = Config.Get("data-directory", "../../../Data");
+            var destinationFolder = Path.Combine(dataDirectory, "trading-economics");
```
Comment (jaredbroad) [inline ToolBox/TradingEconomicsDataDownloader/TradingEconomicsEarningsDownloaderProgram.cs]: `/alternative/..`

File: ToolBox/TradingEconomicsDataDownloader/TradingEconomicsIndicatorDownloader.cs [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        private readonly string _destinationFolder;
+        private readonly DateTime _fromDate;
+        private readonly DateTime _toDate;
+        private string _indicator;
+
+        public TradingEconomicsIndicatorDownloader(DateTime fromDate, DateTime toDate, string destinationFolder)
+        {
+            _fromDate = fromDate;
+            _toDate = toDate;
+            _destinationFolder = Path.Combine(destinationFolder, "world", "daily");
```
Comment (jaredbroad) [inline ToolBox/TradingEconomicsDataDownloader/TradingEconomicsIndicatorDownloader.cs]: per comment above

File: ToolBox/TradingEconomicsDataDownloader/TradingEconomicsIndicatorDownloaderProgram.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    public static class TradingEconomicsIndicatorDownloaderProgram
+    {
+        /// <summary>
+        /// Trading Economics Calendar Downloader Toolbox Project For LEAN Algorithmic Trading Engine.
+        /// </summary>
+        public static void TradingEconomicsIndicatorsDownloader(DateTime fromDate, DateTime toDate)
+        {
+            // Load settings from config.json
+            var dataDirectory = Config.Get("data-directory", "../../../Data");
+            var destinationFolder = Path.Combine(dataDirectory, "trading-economics");
```
Comment (jaredbroad) [inline ToolBox/TradingEconomicsDataDownloader/TradingEconomicsIndicatorDownloaderProgram.cs]: per comments above

**Pull request merged**